### DISCUSSION TITLE
Using pymorthy2

### DISCUSSION THREAD

**Discussion by quillcraftsman** [findsimilar/find-similar#21] [General]
We have https://github.com/findsimilar/demo/issues/14 this issue in demo project about comparability `python 3.11` and `pymorthy2` (`getargspec` was removed in `python 3.11)
And we already have a patch https://github.com/pymorphy2/pymorphy2/pull/167

There are several problems here:
* Last commit in `pymorthy2` was 3 years ago
* This library works only with Russian and Ukrainian languages

If pull request with patch from @HammadRafique29 will be accepted. There is no problems then.
But if not. We well need to decide what to do.

Now I see this ways:
* Don't use `pymorphy2`
* Find a supported fork
* Fork or copy it ourselves (there is no licence in `pymorphy2` repository)

I propose to wait some time (a week may be). If pull request with patch will not confirm, discuss this issue here.

**Comment by HammadRafique29** [comment]
Yes, let's wait for week or 2. Otherwise we should implement own code

**Comment by quillcraftsman** [comment]
I am finking about: If we get this package and make our `pymorpy3` (or something like this) we can add more languages and new functional. But it's a lot of work there.

**Reply by HammadRafique29** [reply]
Yes, it take some time but not impossible

**Comment by quillcraftsman** [comment]
We can check this https://github.com/no-plagiarism/pymorphy3. It's most actual fork (last updates 2 days ago). It's already in pypi `pymorphy3`. But I don't know what about `python 3.11` support. 

 @HammadRafique29 , do you want to check it and install instead of `pymorphy2`? Or If it don't works with `python 3.11` try to make pull request there.

**Reply by HammadRafique29** [reply]
Yes, i have checked it. Pymorphy3 also have the same problem like Pymorphy2

**Reply by HammadRafique29** [reply]
![image](https://github.com/findsimilar/find-similar/assets/112252001/fbf5d265-9dc4-47e2-a48f-f2af998d7b7b)

**Reply by quillcraftsman** [reply]
I think `pymorphy3` has more chances to accept pull request then `pymorphy2`

**Reply by HammadRafique29** [reply]
So i have to pull request for pymorphy3

**Reply by HammadRafique29** [reply]
@quillcraftsman I have tested the `pymorphy3`, and somehow the code is running. No need to pull request for `pymorphy3`.

The only thing we have to doo is:

- Go to `find_simple/tokenize.py` [here](https://github.com/findsimilar/find-similar/blob/main/find_similar/tokenize.py)
- Replace all the `pymorphy2` to `pymorphy3 `
- Replace`pymorphy2==0.9.1`  and  `pymorphy2-dicts-ru==2.4.417127.4579844` from requirement to `pymorphy3==1.2.0`


**Problem Solved!**

**Comment by quillcraftsman** [comment]
> @quillcraftsman I have tested the `pymorphy3`, and somehow the code is running. No need to pull request for `pymorphy3`.
> 
> The only thing we have to doo is:
> 
> * Go to `find_simple/tokenize.py` [here](https://github.com/findsimilar/find-similar/blob/main/find_similar/tokenize.py)
> * Replace all the `pymorphy2` to `pymorphy3 `
> * Replace`pymorphy2==0.9.1`  and  `pymorphy2-dicts-ru==2.4.417127.4579844` from requirement to `pymorphy3==1.2.0`
> 
> **Problem Solved!**

Sounds great. I propose to do this and add `python 3.11` check to GitHub test action (we will run test on 3.10 and 3.11). I will create a special issue for this task soon.

**Comment by quillcraftsman** [comment]
> > @quillcraftsman I have tested the `pymorphy3`, and somehow the code is running. No need to pull request for `pymorphy3`.
> > The only thing we have to doo is:
> > 
> > * Go to `find_simple/tokenize.py` [here](https://github.com/findsimilar/find-similar/blob/main/find_similar/tokenize.py)
> > * Replace all the `pymorphy2` to `pymorphy3 `
> > * Replace`pymorphy2==0.9.1`  and  `pymorphy2-dicts-ru==2.4.417127.4579844` from requirement to `pymorphy3==1.2.0`
> > 
> > **Problem Solved!**
> 
> Sounds great. I propose to do this and add `python 3.11` check to GitHub test action (we will run test on 3.10 and 3.11). I will create a special issue for this task soon.

https://github.com/findsimilar/find-similar/issues/23 has been created

**Comment by quillcraftsman** [comment]
The solution: https://github.com/findsimilar/find-similar/pull/28. Waiting for accept.

**Comment by quillcraftsman** [comment]
Resolved in `find-similar==1.2.1`, [find-similar repository](https://github.com/findsimilar/find-similar), [demo repository](https://github.com/findsimilar/demo)
